File: app/src/main/java/com/unis/gameplatfrom/presenter/LoginPresenter.java
package com.unis.gameplatfrom.presenter;

import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.widget.Toast;

import com.trello.rxlifecycle2.android.FragmentEvent;
import com.unis.gameplatfrom.api.HUDLoadDataSubscriber;
import com.unis.gameplatfrom.api.result.BaseObjectResult;
import com.unis.gameplatfrom.cache.UserCenter;
import com.unis.gameplatfrom.constant.LoginConstant;
import com.unis.gameplatfrom.entity.LoginEntity;
import com.unis.gameplatfrom.model.LoginModel;
import com.unis.gameplatfrom.ui.MainActivity;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class LoginPresenter extends BasePresenter<LoginConstant.Model,LoginConstant.View> implements LoginConstant.Presenter{

    private Context context;

    public LoginPresenter(Context context) {
        this.context = context;
    }

    @Override
    public void login(String account, String password, String netaddress, int type, String key) {

        if(isViewAttached()) {
            getModule().login(account, password, netaddress, type, key)
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    //.compose(this.bindUntilEvent(FragmentEvent.DESTROY))
                    .subscribe(new HUDLoadDataSubscriber<LoginEntity>(context) {
                        @Override
                        public void onNext(LoginEntity result) {

                            if (result.getErr() == 0) {

                                //保存用户数据
                                UserCenter.getInstance().setUserName(result.getName());
                                UserCenter.getInstance().setUserHead(result.getHead());
                                UserCenter.getInstance().setToken(result.getUuid());
                                getView().onSuccess(result);

                            }
                            if (result.getErr() == 1) {
                                Toast.makeText(context, result.getMsg(), Toast.LENGTH_SHORT).show();
                            }
                        }
                    });
        }
}

    @Override
    protected LoginConstant.Model createModule() {
        return  new LoginModel();
    }

    @Override
    public void start() {

    }
}

File: kotlin/src/main/java/com/unis/kotlin/presenter/LoginPresenter.kt
package com.unis.kotlin.presenter

import android.content.Context
import android.widget.Toast
import com.unis.kotlin.api.HUDLoadDataSubscriber
import com.unis.kotlin.cache.UserCenter
import com.unis.kotlin.constant.LoginConstant
import com.unis.kotlin.entity.LoginEntity
import com.unis.kotlin.model.LoginModel
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

class LoginPresenter : BaesPresenter<LoginConstant.Model, LoginConstant.View>,LoginConstant.Presenter{

    private var context: Context? = null;

    constructor(context: Context){
        this.context = context;
    }

    override fun login(account: String, password: String, netaddress: String, type: Int, key: String) {

        if (isViewAttached()) {
            getModule()!!.login(account, password, netaddress, type, key)
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    //.compose(this.bindUntilEvent(FragmentEvent.DESTROY))
                    .subscribe(object : HUDLoadDataSubscriber<LoginEntity>(this.context!!) {
                        override fun onNext(result: LoginEntity) {

                            if (result.err == 0) {

                                //保存用户数据
                                UserCenter.get()!!.setUserName(result.name)
                                UserCenter.get()!!.setUserHead(result.head)
                                UserCenter.get()!!.setToken(result.uuid)
                                getView()!!.onSuccess(result)

                            }
                            if (result.err == 1) {
                                Toast.makeText(context, result.msg, Toast.LENGTH_SHORT).show()
                            }
                        }
                    });
    }



    }


    override fun createModule(): LoginConstant.Model {
        return LoginModel()
    }

    override fun start() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
}